packages/web/docs/src/content/gateway/logging-and-error-handling.mdx

```diff
@@ -0,0 +1,305 @@
+| `UNAUTHENTICATED`           | The given auth credentials are not valid.<br/>Check the logs, and documentation of the used auth plugin to learn more                                                                                                                                                                                    |
+| `PERSISTED_QUERY_NOT_FOUND` | Indicates that persisted operation information is not found in the store.<br/>Check the related persisted operation plugin docs to learn more about this error                                                                                                                                           |
+| `INTERNAL_SERVER_ERROR`     | Indicates that the error is unexpected/unspecified and masked by the gateway.<br/>It is probably caused by an unexpected network, connection or any other runtime error.<br/>You can see the details of this error in the logs.                                                                          |
+| `DOWNSTREAM_SERVICE_ERROR`  | Indicates the error is subgraph related, and generated by the subgraph not the gateway                                                                                                                                                                                                                   |
+
+### Error Masking
+
+Hive Gateway masks internal server errors by default to prevent leaking sensitive information to the
+client. But without any codes all errors in the result are considered safe from the subgraph.
+
+Understanding this concept is crucial for building secure applications.
+
+So any HTTP errors, network errors, or any other errors that are not related to the subgraph are
+masked by default. But any errors sent by the subgraph are not masked by default.
+
+All masked errors are replaced with a generic error message and the original error is not exposed to
+the client.
+
+```json
+{
+  "errors": [
+    {
+      "message": "Unexpected error.",
+      "code": "INTERNAL_SERVER_ERROR",
+      "locations": [
+        {
+          "line": 2,
+          "column": 3
+        }
+      ],
+      "path": ["greeting"]
+    }
+  ],
+  "data": null
+}
+```
+
+But if the subgraph sends an error in the result, it is forwarded with `DOWNSTREAM_SERVICE_ERROR` if
+`INTERNAL_SERVER_ERROR` code isn't passed.
+
+```json
+{
+  "errors": [
+    {
+      "message": "This error is from subgraph",
+      "code": "DOWNSTREAM_SERVICE_ERROR",
+      "locations": [
+        {
+          "line": 2,
+          "column": 3
+        }
+      ],
+      "path": ["greeting"]
+    }
+  ],
+  "data": null
+}
+```
+
+<Callout>
+  When `INTERNAL_SERVER_ERROR` is passed from the subgraph, it is masked by the gateway and sent to
+  the client as `INTERNAL_SERVER_ERROR`. But the gateway will still log the original error.
+</Callout>
+
+#### Disable for development
```

Comment:
I think development should actually be the same as prod. This is also what we do in Hive. 
Because of the frontend developer assumes some structure during it, it will be different in prod. 

@n1ru4l @kamilkisiela what do you think?

Reply:
It is already same by default in development. This section shows how to disable error masking for debugging purposes.

Reply:
maybe we should note that this should be disabled only for debugging? and not for actual development? not sure how to phrase it 🤔

Reply:
I copied it from Yoga docs actually :/

Reply:
Improved it!